1.Exploratory_Data_Analysis/scripts/2c.correlation_nomic_and_agg_morphology.py

```diff
@@ -0,0 +1,245 @@
+
+
+# remove uM in each row of the Metadata_inducer1_concentration column if it is present
+if "Metadata_inducer1_concentration" in feature_df.columns:
+    feature_df["Metadata_inducer1_concentration"] = feature_df[
+        "Metadata_inducer1_concentration"
+    ].str.replace("µM", "")
+# replace nan values with 0
+feature_df["Metadata_inducer1_concentration"] = feature_df[
+    "Metadata_inducer1_concentration"
+].fillna(0)
+feature_df["Metadata_inducer2_concentration"] = feature_df[
+    "Metadata_inducer2_concentration"
+].fillna(0)
+feature_df["Metadata_inhibitor_concentration"] = feature_df[
+    "Metadata_inhibitor_concentration"
+].fillna(0)
+# treatment column merge
+conditions = [
+    (feature_df["Metadata_inducer2"].isnull()),
+    feature_df["Metadata_inducer2"].notnull(),
+]
+results = [
+    (feature_df["Metadata_inducer1"]).astype(str),
+    (feature_df["Metadata_inducer1"] + "_" + feature_df["Metadata_inducer2"]).astype(
+        str
+    ),
+]
+feature_df["Metadata_Treatment"] = np.select(condlist=conditions, choicelist=results)
+
+# dose column merge
+conditions = [
+    (feature_df["Metadata_inducer2"].isnull()),
+    feature_df["Metadata_inducer2"].notnull(),
+]
+
+results = [
+    (feature_df["Metadata_inducer1_concentration"].astype(str)).astype(str),
+    (
+        feature_df["Metadata_inducer1_concentration"].astype(str)
+        + "_"
+        + feature_df["Metadata_inducer2_concentration"].astype(str)
+    ).astype(str),
+]
+feature_df["Metadata_Dose"] = np.select(condlist=conditions, choicelist=results)
+feature_df["Metadata_inducer1_concentration"] = pd.to_numeric(
+    feature_df["Metadata_inducer1_concentration"]
+)
+# one beta of inudcer1, inducer1 concentration, inhibitor, and inhibitor concentration all as 1 beta term
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = (
+    feature_df["Metadata_Treatment"]
+    + "_"
+    + feature_df["Metadata_Dose"].astype(str)
+    + "_"
+    + feature_df["Metadata_inhibitor"].astype(str)
+    + "_"
+    + feature_df["Metadata_inhibitor_concentration"].astype(str)
+).astype(str)
+
+
+# ## Clean up Nomic df
+
+# In[5]:
+
+
+# remove column if colname has pgml in it
+nomic_df = nomic_df.loc[:, ~nomic_df.columns.str.contains("pgml", case=False, na=False)]
+# if column does not contain [NSU] then prefix with Metadata_
+for col in nomic_df.columns:
+    if not any(x in col for x in ["NSU"]):
+        nomic_df = nomic_df.rename(columns={col: "Metadata_" + col})
+
+
+# In[6]:
+
+
+# drop first column of metadata
+nomic_df.columns[3:25]
+nomic_df = nomic_df.drop(nomic_df.columns[3:25], axis=1)
+nomic_df = nomic_df.drop(nomic_df.columns[0:2], axis=1)
+
+
+# ### Merge Nomic and Aggregated Morphology dfs
+
+# In[7]:
+
+
+# merge nomic and morphology data by metadata_well and position_x
+merged_df = pd.merge(
+    feature_df,
+    nomic_df,
+    left_on=["Metadata_Well"],
+    right_on=["Metadata_position_x"],
+)
+merged_df = merged_df.drop(["Metadata_position_x"], axis=1)
```

Comment:
```suggestion
merged_df = pd.merge(
    feature_df,
    nomic_df,
    left_on=["Metadata_Well"],
    right_on=["Metadata_position_x"],
).drop(["Metadata_position_x"], axis=1)
```

Welcome to pandas chaining!